### PR TITLE
virtio_serial: Hotplug pci and chardev and port orderly

### PR DESCRIPTION
**Pull request by nanliu-r**
ID: 1658927

Signed-off-by: Nana Liu <nanliu@redhat.com>

### CODE REVIEW

**Comment by nanliu-r** [comment]
Test PASS

 (1/2) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-
qemu.virtio_port_hotplug.hotplug_port_pci.with_chardev.q35: PASS (64.95 s)
 (2/2) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.virtio_port_hotplug.hotplug_port_pci.without_chardev.q35: PASS (54.66 s)

 (1/2) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_port_hotplug.hotplug_port_pci.with_chardev.q35: PASS (137.65 s)
 (2/2) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_port_hotplug.hotplug_port_pci.without_chardev.q35: PASS (122.64 s)

**Comment by nanliu-r** [comment]
@vivianQizhu  Please help to review this, thanks a lot.

**Comment by nanliu-r** [comment]
> @vivianQizhu Please help to review this, thanks a lot.

Updated and test pass, please help to review this, thanks.

 (1/2) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.virtio_port_hotplug.hotplug_port_pci.with_chardev: PASS (59.14 s)
 (2/2) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.virtio_port_hotplug.hotplug_port_pci.without_chardev: PASS (49.99 s)

File: qemu/tests/virtio_serial_hotplug_port_pci.py [hunk 1]
```diff
@@ -13,6 +13,21 @@
 from qemu.tests.vioser_in_use import live_migration_guest  # pylint: disable=W0611
 
 
+def get_buses_serial_devices(vm, params, char_devices, serials):
```
Comment (vivianQizhu) [inline qemu/tests/virtio_serial_hotplug_port_pci.py]: 1. Recommend `get_buses_and_serial_devices` to be clearer.
2. Doc string please.

**Comment by vivianQizhu** [comment]
Ack

**Comment by vivianQizhu** [comment]
@nanliu-r Test result please, for all test cases being influenced. Thanks.

**Comment by nanliu-r** [comment]
Test Pass.

 (2/9) Host_RHEL.m8.u2.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_port_hotplug.hotplug_port_pci: PASS (84.94 s)
 (3/9) Host_RHEL.m8.u2.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_port_hotplug.hotplug_port_pci.with_reboot: PASS (139.41 s)
 (4/9) Host_RHEL.m8.u2.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_port_hotplug.hotplug_port_pci.with_system_reset: PASS (143.95 s)
 (5/9) Host_RHEL.m8.u2.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_port_hotplug.hotplug_port_pci.with_shutdown_after_unplug: PASS (62.77 s)
 (6/9) Host_RHEL.m8.u2.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_port_hotplug.hotplug_port_pci.with_shutdown_after_plug:PASS (56.81 s)
 (7/9) Host_RHEL.m8.u2.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_port_hotplug.hotplug_port_pci.with_live_migration_after_unplug: PASS (71.96 s)
 (8/9) Host_RHEL.m8.u2.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_port_hotplug.hotplug_port_pci.repeat_pci_in_loop: PASS (1303.75 s)
 (9/9) Host_RHEL.m8.u2.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_port_hotplug.hotplug_port_pci.without_chardev: PASS (58.21 s)

(1/1) Host_RHEL.m8.u2.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_port_hotplug.hotplug_port_pci.q35: PASS (96.52 s)

**Pull request merged**